types/mempool/mempool.go

```diff
@@ -1,38 +1,39 @@
 package mempool
 
 import (
-	"github.com/cosmos/cosmos-sdk/types"
-)
-
-// Tx we define an app-side mempool transaction interface that is as
-// minimal as possible, only requiring applications to define the size of the
-// transaction to be used when reaping and getting the transaction itself.
-// Interface type casting can be used in the actual app-side mempool implementation.
-type Tx interface {
-	types.Tx
-
-	// Size returns the size of the transaction in bytes.
-	Size() int
+	"errors"
 
-	// Hash returns the hash of the transaction.
-	Hash() [32]byte
-}
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
 
 type Mempool interface {
 	// Insert attempts to insert a Tx into the app-side mempool returning
 	// an error upon failure.
-	Insert(types.Context, Tx) error
+	Insert(sdk.Context, sdk.Tx) error
 
-	// Select returns the next set of available transactions from the app-side
-	// mempool, up to maxBytes or until the mempool is empty. The application can
-	// decide to return transactions from its own mempool, from the incoming
-	// txs, or some combination of both.
-	Select(ctx types.Context, txs [][]byte, maxBytes int) ([]Tx, error)
+	// Select returns an Iterator over the app-side mempool. If txs are specified,
+	// then they shall be incorporated into the Iterator. The Iterator must
```

Comment:
In our implementation, the transactions are not incorporated but just ignored. Is that expected?

Reply:
Correct, Osmosis (and simapp) just ignore them and use the FIFO-ordered txs that are sent from Tendermint.